src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -499,26 +476,63 @@ private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt)
         isCurrentlyUploading.set(false);
     }
 
+    private void deleteFilesInComponents(Map<String, Set<String>> fileDeleteSet) {
+        for (Map.Entry<String, Set<String>> entry : fileDeleteSet.entrySet()) {
+            String key = entry.getKey();
+            Set<String> fileNames = entry.getValue();
+            fileNames.stream().map(LogFile::new).forEach(file -> {
+                if (!lastComponentUploadedLogFileInstantMap.containsKey(key)
+                        || lastComponentUploadedLogFileInstantMap.get(key)
+                        .isBefore(Instant.ofEpochMilli(file.lastModified()))) {
+                    lastComponentUploadedLogFileInstantMap.put(key, Instant.ofEpochMilli(file.lastModified()));
+                }
+                // delete the entry if one file is read and uploaded completely
+                fileHashNamePairs.remove(file.hashString());
```

Comment:
if we remove it here, but do _not_ delete the file from the disk, won't we upload it again? We wouldn't want to do that, right?